Assets/Scripts/PlayerController.cs

```diff
@@ -1,644 +1,544 @@
-	public AudioClip powerupSound;
-
-	public AudioClip speedUpSFX1;
-	public AudioClip speedUpSFX2;
-	public AudioClip speedUpSFX3;
-
-	public AudioClip sizeUpSFX1;
-	public AudioClip sizeUpSFX2;
-
-	public AudioClip pandemoniumSFX1;
-	public AudioClip pandemoniumSFX2;
-
-	public Material masterColor;
-	public Material player1Material;
-	public Material player2Material;
-	public Material player3Material;
-	public Material player4Material;
-
-	public TextMesh pandemoniumCounter;
-
-	public GameObject trail;
-
-	public float penaltyTimer;
-	private bool penaltyTimerActive = false;
-
-	private float speedPowerupTimer;
-	private bool speedPowerupActive = false; 
-
-	private float sizePowerupTimer;
-	private bool sizePowerupActive = false; 
-
-	private float pandemoniumTimer;
-	private bool pandemoniumPowerupActive = false;
-
-	Rigidbody2D rb;
-	MeshRenderer mr;
-
-	// Use this for initialization
-	void Start () {
-
-		gravButton_Xbox = gravButton + "_Xbox";
-		jumpButton_Xbox = jumpButton + "_Xbox";
-		horiz_Xbox = horiz + "_Xbox";
-
-		ps = transform.Find("ssps").GetComponent<ParticleSystem> ();
-		ps.Stop ();
-		rb = GetComponent<Rigidbody2D>();
-		mr = GetComponent<MeshRenderer> ();
-		PolygonCollider2D pg = GetComponent<PolygonCollider2D> ();
-		rb.gravityScale = startingGrav;
-		startMass = rb.mass;
-		canMove = true;
-		pandemoniumCounter.GetComponent<TextMesh> ().color = new Vector4(0f, 0f, 0f, 0f);
-
-		//GetComponent<MeshRenderer> ().material = masterColor;
-		inPenalty = false;
-		// assign player color
-		switch (playerID) {
-		case 1:
-			playerColor = "1069A8";
-			break;
-		case 2:
-			playerColor = "7CBEE8";
-			break;
-		case 3:
-			playerColor = "D63236";
-			break;
-		case 4:
-			playerColor = "D97A7B";
-			break;
+﻿using UnityEngine;
+using System.Collections;
+using PigeonCoopToolkit.Effects.Trails;
+
+public class PlayerController : MonoBehaviour {
+
+    // Core shape stats, public for tesitng 
+    public float jumpPower;
+    public float speed;
+	public float spinPower;
+
+    // Save original stats for reversion from powerups
+	private float startSpeed;
+	private float startMass;
+	private float startJumpPower;
+
+    // Button names
+	public string horiz = "Horizontal_P1";
+    public string jumpButton = "Jump_P1";
+    public string gravButton = "Grav_P1";
+	private string horiz_Xbox;
+	private string jumpButton_Xbox;
+	private string gravButton_Xbox;
+
+    // Properties of player by ID
+	public int playerID;
+	public int playerType = 0;
+    private string playerColor;
+
+    // Misc initializations
+    public int team = 1;
+    public float startingGrav = 1;
+    public bool isJumping = false;
+    private bool inPenalty = false;
+    private bool canMove = true;
+
+    // Particle system
+	public ParticleSystem ps;
+
+    // Rigidbody, mesh, colliders
+    Rigidbody2D rb;
+    MeshRenderer mr;
+    public PolygonCollider2D trianglePC, trapezoidPC, rectanglePC, starPC;
+    private Collider2D shapeCollider;

```

Comment:
Yeah! PolygonCollider2D, CircleCollider2D and BoxCollider2D all inherit from it, so you can cast them as a Collider2D variable and call their shared functions without the compiler complaining.